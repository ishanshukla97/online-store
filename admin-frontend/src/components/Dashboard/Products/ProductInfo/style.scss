@import "../../../../sass/main";

.product-form {
	position: relative;
	width: 50%;
	height: 45rem;
	margin: 1rem;
	@include box-shadow-1;
	border-radius: 5px;
	padding: 4rem;
	background-color: $color-grey-light-1;

	&__input {
		label {
			font-size: 1rem;
		}
	}

	&__action {
		width: 100%;
		height: 8rem;
		padding: 2rem;

		&--cancel {
			float: left;
			background-color: black;
			transition: all 0.2s;
			&:hover {
				background-color: rgb(145, 0, 0);
			}
		}
		&--delete {
			float: right;
			background-color: red;
			margin-right: 1rem;
			transition: all 0.2s;
			&:hover {
				background-color: rgb(131, 23, 23);
			}
		}
		&--update,
		&--create {
			float: right;
			background-color: rgb(8, 248, 0);
			margin-right: 1rem;
			transition: all 0.2s;
			&:hover {
				background-color: rgb(0, 128, 43);
			}
		}
	}
}
