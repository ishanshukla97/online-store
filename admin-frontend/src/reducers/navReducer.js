import { MAIN_VIEW } from "../actions/types";

export default function(state = {}, action) {
	switch (action.type) {
		case MAIN_VIEW:
			return action.payload;
		default:
			return state;
	}
}
