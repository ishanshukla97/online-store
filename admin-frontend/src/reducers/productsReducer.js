import { FETCH_PRODUCTS } from "../actions/types";

export default function(state = [{}], actions) {
	switch (actions.type) {
		case FETCH_PRODUCTS:
			return actions.payload.data.data.products || false;
		default:
			return state;
	}
}
