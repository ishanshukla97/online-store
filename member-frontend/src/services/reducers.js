import { combineReducers } from "redux";
import { reducer as formReducer } from "redux-form";
import deskReducer from "./desk/reducer";
import filterReducer from "./filter/reducer";

export default combineReducers({
	desk: deskReducer,
	filterDesk: filterReducer,
	form: formReducer
});
