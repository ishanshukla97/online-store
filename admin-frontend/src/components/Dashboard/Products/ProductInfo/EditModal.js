import React from "react";
import { reduxForm, Field } from "redux-form";
import "./style.scss";

const ProductEdit = props => {
	const itemKeys = Object.keys(props.item);
	let isNew;
	const renderSubmit = () => {
		if (props.item._id) {
			isNew = false;
			return (
				<button
					type="submit"
					className="btn product-form__action--update"
				>
					Update
				</button>
			);
		}
		isNew = true;
		return (
			<button type="submit" className="btn product-form__action--create">
				Create
			</button>
		);
	};

	const renderForm = () => {
		return (
			<div className="product-form">
				<form
					onSubmit={props.handleSubmit(values =>
						props.onSubmit(values)
					)}
				>
					{itemKeys.map(key => {
						return (
							<Field
								key={key}
								type="text"
								name={key}
								label={key.toUpperCase()}
								component={ModalField}
							/>
						);
					})}
					<div className="product-form__action">
						<button
							className="btn product-form__action--cancel"
							onClick={() => props.onCancel()}
						>
							Cancel
						</button>
						{renderSubmit()}
						<button
							className="btn product-form__action--delete"
							onClick={() => props.onRemove(props.item)}
							disabled={isNew}
						>
							Delete
						</button>
					</div>
				</form>
			</div>
		);
	};

	return renderForm();
};

const ModalField = props => {
	const { input, label, type, meta } = props;

	const isId = label === "_ID" ? true : false;
	return (
		<div className="product-form__input">
			<label>{label}</label>
			<input type={type} {...input} disabled={isId} />
			<div className="red-text">{meta.touched ? meta.error : ""}</div>
		</div>
	);
};

function validate(values) {
	const errors = {};

	//Implement form validation
	if (!values.title) {
		errors.title = "Title cannot be empty";
	}
	if (!values.description) {
		errors.description = "Description cannot be empty";
	}
	if (!values.price) {
		errors.price = "Price cannot be empty";
	}

	if (values.price && /[a-zA-Z]/.test(values.price)) {
		errors.price = "Price must not contain any alphabet";
	}
	if (!values.img) {
		errors.img = "You must specify image name";
	}
	return errors;
}

export default reduxForm({ validate, form: "editProductForm" })(ProductEdit);
