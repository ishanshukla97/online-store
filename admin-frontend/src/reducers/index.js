import { combineReducers } from "redux";
import authReducer from "./authReducer";
import { reducer as formReducer } from "redux-form";
import navReducer from "./navReducer";
import productsReducer from "./productsReducer";

// keys inside of combineReducers object represent
// keys in the state object
export default combineReducers({
	auth: authReducer,
	form: formReducer,
	view: navReducer,
	products: productsReducer
});
