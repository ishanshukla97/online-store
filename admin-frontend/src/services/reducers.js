import { combineReducers } from "redux";
import { reducer as formReducer } from "redux-form";
import viewReducer from "./view/reducer";
import authReducer from "./auth/reducer";
import productsReducer from "./products/reducer";

// keys inside of combineReducers object represent
// keys in the state object
export default combineReducers({
	auth: authReducer,
	form: formReducer,
	view: viewReducer,
	products: productsReducer
});
