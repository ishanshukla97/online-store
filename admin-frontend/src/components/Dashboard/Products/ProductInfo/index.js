import React from "react";
import EditModal from "./EditModal";

const ProductInfo = props => {
	let product = props.products.filter(product => product._id === props.id)[0];

	if (!product || product === {}) {
		let newProduct = Object.assign({}, props.products[1]);

		Object.keys(newProduct).forEach(k => (newProduct[k] = ""));

		return (
			<EditModal
				item={newProduct}
				initialValues={{ ...newProduct }}
				onCancel={() => props.onCancel()}
				onSubmit={values => props.onSubmit(values)}
				onRemove={values => props.onRemove(values)}
			/>
		);
	}

	return (
		<EditModal
			item={product}
			initialValues={{ ...product }}
			onCancel={() => props.onCancel()}
			onSubmit={values => props.onSubmit(values)}
			onRemove={values => props.onRemove(values)}
		/>
	);
};

export default ProductInfo;
