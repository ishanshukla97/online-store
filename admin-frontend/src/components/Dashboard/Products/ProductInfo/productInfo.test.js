import React from "react";
import renderer from "react-test-renderer";
import ProductInfo from "./index";

import { Provider } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "../../../../reducers";

const store = createStore(reducers, {}, applyMiddleware(reduxThunk));

const mockItemList = [
	{
		_id: 0,
		title: "Watch"
	},
	{
		_id: 1,
		title: "Clock"
	}
];

const productEditInfo = renderer.create(
	<Provider store={store}>
		<ProductInfo products={mockItemList} id={0} />
	</Provider>
);
const productNewInfo = renderer.create(
	<Provider store={store}>
		<ProductInfo products={mockItemList} id={false} />
	</Provider>
);

describe("Product Info", () => {
	it("renders edit product form", () => {
		expect(productEditInfo).toMatchSnapshot();
	});
	it("renders new product form", () => {
		expect(productNewInfo).toMatchSnapshot();
	});
});
