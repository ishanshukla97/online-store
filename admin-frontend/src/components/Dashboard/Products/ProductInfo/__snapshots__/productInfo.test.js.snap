// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Info renders edit product form 1`] = `
<div
  className="container card"
  style={
    Object {
      "marginLeft": "10%",
      "position": "fixed",
      "width": "60%",
    }
  }
>
  <form
    className="card-content"
    onSubmit={[Function]}
  >
    <div>
      <label>
        _ID
      </label>
      <input
        disabled={true}
        name="_id"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "marginBottom": "5px",
          }
        }
        type="text"
        value={0}
      />
      <div
        className="red-text"
        style={
          Object {
            "marginBottom": "5	px",
          }
        }
      >
        
      </div>
    </div>
    <div>
      <label>
        TITLE
      </label>
      <input
        disabled={false}
        name="title"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "marginBottom": "5px",
          }
        }
        type="text"
        value="Watch"
      />
      <div
        className="red-text"
        style={
          Object {
            "marginBottom": "5	px",
          }
        }
      >
        
      </div>
    </div>
    <div
      className="card-action"
      style={
        Object {
          "marginBottom": "30px",
        }
      }
    >
      <button
        className="btn-flat red lighten-2 left"
        onClick={[Function]}
      >
        Cancel
      </button>
      <button
        className="btn-flat teal right"
        style={
          Object {
            "marginLeft": "10px",
          }
        }
        type="submit"
      >
        Create
      </button>
      <button
        className="btn-flat red darken-2 right"
        disabled={true}
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </form>
</div>
`;

exports[`Product Info renders new product form 1`] = `
<div
  className="container card"
  style={
    Object {
      "marginLeft": "10%",
      "position": "fixed",
      "width": "60%",
    }
  }
>
  <form
    className="card-content"
    onSubmit={[Function]}
  >
    <div>
      <label>
        _ID
      </label>
      <input
        disabled={true}
        name="_id"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "marginBottom": "5px",
          }
        }
        type="text"
        value={0}
      />
      <div
        className="red-text"
        style={
          Object {
            "marginBottom": "5	px",
          }
        }
      >
        
      </div>
    </div>
    <div>
      <label>
        TITLE
      </label>
      <input
        disabled={false}
        name="title"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "marginBottom": "5px",
          }
        }
        type="text"
        value="Watch"
      />
      <div
        className="red-text"
        style={
          Object {
            "marginBottom": "5	px",
          }
        }
      >
        
      </div>
    </div>
    <div
      className="card-action"
      style={
        Object {
          "marginBottom": "30px",
        }
      }
    >
      <button
        className="btn-flat red lighten-2 left"
        onClick={[Function]}
      >
        Cancel
      </button>
      <button
        className="btn-flat teal right"
        style={
          Object {
            "marginLeft": "10px",
          }
        }
        type="submit"
      >
        Create
      </button>
      <button
        className="btn-flat red darken-2 right"
        disabled={true}
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </form>
</div>
`;

exports[`renders edit product form 1`] = `
<div
  className="container card"
  style={
    Object {
      "marginLeft": "10%",
      "position": "fixed",
      "width": "60%",
    }
  }
>
  <form
    className="card-content"
    onSubmit={[Function]}
  >
    <div>
      <label>
        _ID
      </label>
      <input
        disabled={true}
        name="_id"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "marginBottom": "5px",
          }
        }
        type="text"
        value={0}
      />
      <div
        className="red-text"
        style={
          Object {
            "marginBottom": "5	px",
          }
        }
      >
        
      </div>
    </div>
    <div>
      <label>
        TITLE
      </label>
      <input
        disabled={false}
        name="title"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "marginBottom": "5px",
          }
        }
        type="text"
        value="Watch"
      />
      <div
        className="red-text"
        style={
          Object {
            "marginBottom": "5	px",
          }
        }
      >
        
      </div>
    </div>
    <div
      className="card-action"
      style={
        Object {
          "marginBottom": "30px",
        }
      }
    >
      <button
        className="btn-flat red lighten-2 left"
        onClick={[Function]}
      >
        Cancel
      </button>
      <button
        className="btn-flat teal right"
        style={
          Object {
            "marginLeft": "10px",
          }
        }
        type="submit"
      >
        Create
      </button>
      <button
        className="btn-flat red darken-2 right"
        disabled={true}
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </form>
</div>
`;

exports[`renders new product form 1`] = `
<div
  className="container card"
  style={
    Object {
      "marginLeft": "10%",
      "position": "fixed",
      "width": "60%",
    }
  }
>
  <form
    className="card-content"
    onSubmit={[Function]}
  >
    <div>
      <label>
        _ID
      </label>
      <input
        disabled={true}
        name="_id"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "marginBottom": "5px",
          }
        }
        type="text"
        value={0}
      />
      <div
        className="red-text"
        style={
          Object {
            "marginBottom": "5	px",
          }
        }
      >
        
      </div>
    </div>
    <div>
      <label>
        TITLE
      </label>
      <input
        disabled={false}
        name="title"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "marginBottom": "5px",
          }
        }
        type="text"
        value="Watch"
      />
      <div
        className="red-text"
        style={
          Object {
            "marginBottom": "5	px",
          }
        }
      >
        
      </div>
    </div>
    <div
      className="card-action"
      style={
        Object {
          "marginBottom": "30px",
        }
      }
    >
      <button
        className="btn-flat red lighten-2 left"
        onClick={[Function]}
      >
        Cancel
      </button>
      <button
        className="btn-flat teal right"
        style={
          Object {
            "marginLeft": "10px",
          }
        }
        type="submit"
      >
        Create
      </button>
      <button
        className="btn-flat red darken-2 right"
        disabled={true}
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </form>
</div>
`;
